third_party/move/move-compiler-v2/src/env_pipeline/inliner.rs

```diff
@@ -710,11 +820,51 @@ impl<'env, 'rewriter> InlinedRewriter<'env, 'rewriter> {
 
         let lambda_param_map: BTreeMap<Symbol, &Exp> = lambda_args_matched
             .iter()
-            .map(|(param, arg_exp)| (param.0, *arg_exp))
+            .map(|(param, arg_exp)| (param.1 .0, *arg_exp))
             .collect();
 
-        let (regular_params, regular_actuals): (Vec<&Parameter>, Vec<&Exp>) =
+        let mut function_value_map: BTreeMap<usize, Exp> = BTreeMap::new();
+        let mut lifted_lambda_funs: BTreeMap<usize, move_model::model::FunctionData> =
+            BTreeMap::new();
+        let mut sym_para_map: BTreeMap<Symbol, usize> = BTreeMap::new();
+        let mut function_value_spec_map = BTreeMap::new();
+        // Lift lambda expressions and generate corresponding spec functions
+        if lift_inline_funs && target_qualified_fun_id_opt.is_some() {
+            let options = LambdaLiftingOptions {
+                include_inline_functions: true,
+            };
+            let fun_env = env.get_function(target_qualified_fun_id_opt.unwrap());
+            for (para, lambda) in lambda_args_matched.clone() {
+                let mut lifter = LambdaLifter::new(
+                    &options,
+                    &fun_env,
+                    Some(format!(
+                        "_inline_{}_{}",
+                        para.0,
+                        env.get_node_loc(lambda.node_id()).span().start()
+                    )),
+                );
+                let closure_exp = lifter.rewrite_exp(lambda.clone());
+                if lifter.lifted_len() == 1 {
```

Comment:
What does `!= 1` would mean? Should we crash? Add comment.

Reply:
done